api/app/GraphQL/Mutations/DownloadUsersCsv.php

```diff
@@ -24,26 +23,23 @@ public function __invoke($_, array $args)
         $user = Auth::user();
         throw_unless(is_string($user?->id), UnauthorizedException::class);
 
+        $ids = $args['ids'] ?? null;
+        $filters = $args['where'] ?? null;
         $locale = $args['locale'] ?? 'en';
 
         try {
-            // Make sure this user can see candidates before sending
-            // them to the generation job
-            $ids = User::whereIn('id', $args['ids'])
-                ->authorizedToView()
-                ->get('id')
-                ->pluck('id') // Seems weird but we are just flattening it out
-                ->toArray();
-
-            $fileName = sprintf('%s_%s.csv', Lang::get('filename.users', [], $locale), date('Y-m-d_His'));
 
             $generator = new UserCsvGenerator(
-                ids: $ids,
-                fileName: $fileName,
+                fileName: sprintf('%s_%s.csv', Lang::get('filename.users', [], $locale), date('Y-m-d_His')),
                 dir: $user->id,
                 lang: strtolower($locale),
             );
 
+            $generator = $generator
+                ->setUserId($user->id)
+                ->setIds($ids)
+                ->setFilters($filters);
```

Comment:
The constructors were getting kinda fat so I moved out some to builders and made them optional.